terraform/main.tf

```diff
@@ -0,0 +1,117 @@
+data "aws_s3_bucket" "diseasepredictor2025"{
+    bucket = "diseasepredictor2025"
+
+    //tags = {
+    //Owner = element(split("/", data.aws_caller_identity.current.arn), 1)
+//}
+}
+
+data "aws_caller_identity" "current" {}
+
+# Define role
+data "aws_iam_policy_document" "assume_role_policy" {
+  statement {
+    actions = ["sts:AssumeRole"]
+
+    principals {
+      type        = "Service"
+      identifiers = ["ecs-tasks.amazonaws.com"]
+    }
+  }
+}
+# Create role
+resource "aws_iam_role" "execution_role" {
+  name               = "execution_role"
+  //path               = "/system/"
+  assume_role_policy = data.aws_iam_policy_document.assume_role_policy.json
+}
+# Attach pre-defined policy 
+resource "aws_iam_role_policy_attachment" "execution_pre_attach" {
+  role       = aws_iam_role.execution_role.name
+  policy_arn = "arn:aws:iam::aws:policy/service-role/AmazonECSTaskExecutionRolePolicy"
+}
+# Define policy 
+data "aws_iam_policy_document" "execution_policy" {
+  statement {
+    effect    = "Allow"
+    actions   = ["ec2:Describe*"]
+    resources = ["*"]
+  }
+}
+# Create policy
+resource "aws_iam_policy" "execution_policy" {
+  name        = "execution_policy"
+  description = "An execution policy"
+  policy      = data.aws_iam_policy_document.execution_policy.json
+}
+# Attach policy
+resource "aws_iam_role_policy_attachment" "execution_attach" {
+  role       = aws_iam_role.execution_role.name
+  policy_arn = aws_iam_policy.execution_policy.arn
+}
+
+
+# task_role definition 
+data "aws_iam_policy_document" "task_role_policy" {
+  statement {
+    effect = "Allow"
+    actions = ["sts:AssumeRole"]
+
+    principals {
+      type        = "Service"
+      identifiers = ["ecs-tasks.amazonaws.com"]
+    }
+  }
+}
+
+# task_role Create
+resource "aws_iam_role" "task_role" {
+    name = "task_role"
+    assume_role_policy = data.aws_iam_policy_document.task_role_policy.json
+}
+# task_role policy define
+data "aws_iam_policy_document" "task_policy" {
+  statement {
+    effect = "Allow"
+    actions = [
+      "s3:GetObject",
```

Comment:
you will probably also need `PutObject` to upload anything back to S3. You can give `s3:*` permissions to cover your basis since the policy is already restricted to your bucket and application